tests/Entity/JsonPointerTest.php

```diff
@@ -92,4 +92,16 @@ public function getTestData()
 
         );
     }
+
+    public function testJsonPointerWithPropertyPaths()
+    {
+        $initial = new JsonPointer('#/definitions/date');
+        $modified = $initial->withPropertyPaths(array('~definitions/general', '%custom%'));
```

Comment:
@steffkes It depends on which behavior should be tested. Asserting a new object is returned is 1 thing. asserting this has no side effect is another thing (so the original object has **NOT** be changed). I think both can be tested (so proved).